src/foam/core/CSV.js

```diff
@@ -0,0 +1,233 @@
+    },
+
+    function outputPropertyName(o, p, prefix) {
+      if ( ! this.propertyPredicate(o, p) ) return;
+
+      // Checks if property is enum, or object with properties of its own
+      if ( p.of && ( ! foam.core.AbstractEnum.isInstance( o[p.name] ) ) ) {
+        // Appends object name to prefix for CSV Header
+        prefix += p.name + this.nestedObjectSeperator;
+        this.outputPropertyFilteredName(o[p.name], prefix);
+      } else {
+        this.outputPropertyFilteredName(p.name, prefix);
+      }
+    },
+
+    {
+      name: 'outputPropertyFilteredName',
+      code: foam.mmethod({
+        FObject:   function(o, prefix) {
+          // Gets and recurses through object properties
+          var ps = o.cls_.getAxiomsByClass(foam.core.Property);
+
+          for ( var i = 0 ; i < ps.length ; i++ ) {
+            this.outputPropertyName(o, ps[i], prefix);
+          }
+        },
+        Array: function(o) { /* Ignore arrays in CSV */ },
+        Function: function(o) { /* Ignore functions in CSV */ },
+        Object: function(o) { /* Ignore generic objects in CSV */ }
+      }, function(o, prefix) { this.outputHeaderTitle(o, prefix); })
+    },
+
+    function outputProperty(o, p) {
+      if ( ! this.propertyPredicate(o, p) ) return;
+      this.output(o[p.name]);
+    },
+
+    function reset() {
+      this.buf_ = '';
+      return this;
+    },
+
+    function outputPrimitive(val) {
+      this.out(this.buf_.length ? this.delimiter : '', val);
+      return this;
+    },
+
+    {
+      name: 'output',
+      code: foam.mmethod({
+        Undefined: function(o) { this.outputPrimitive(this.undefinedStr); },
+        Null:      function(o) { this.outputPrimitive(null); },
+        String:    function(o) { this.outputPrimitive(o); },
+        FObject:   function(o) {
+          if ( o.outputCSV ) {
+            o.outputCSV(this)
+            return;
+          }
+          
+          // Outputs only the ordinal of ENUM
+          if ( foam.core.AbstractEnum.isInstance(o) ) {
+            this.outputPrimitive(o.ordinal);
+          } else {
+            var ps = o.cls_.getAxiomsByClass(foam.core.Property);
+
+            for ( var i = 0 ; i < ps.length ; i++ ) {
+              this.outputProperty(o, ps[i]);
+            }
+          }
+        },
+        Array: function(o) { /* Ignore arrays in CSV */ },
+        Function: function(o) { /* Ignore functions in CSV */ },
+        Object: function(o) { /* Ignore generic objects in CSV */ }
+      }, function(o) { this.outputPrimitive(o); })
+    },
+
+    function fromCSV(className, s) {
+      if ( ! s ) throw 'Invalid CSV input to convert. Arguments must be (className, csvString).'
+      var lines = s.split('\n');
+
+      if ( lines.length == 0 ) throw 'Insufficient CSV Input'
+
+      // Trims quotes and splits CSV row into array
+      var props = lines[0].split(',');
+
+      // Account for array of values (TODO?)
+      var values = lines[1].split(',');
```

Comment:
There isn't really a way to handle that unless we use a special delimiter (~), or we surround the fields with a symbol ("), as far as I know. What would you recommend?